### PR TITLE
Enabled CORS to allow calls from frontend to the backend api

### LINKED ISSUE #31: Enable CORS in backend
### What happened?
Making a request to the backend from the frontend results in a XMLHTTPError

### How can we reproduce it?
Create a function that hits the GET /ping endpoint in the frontend

https://github.com/gin-contrib/cors

### PR DESCRIPTION
**Pull request by SAbbas2018**
## [DEV-031] Issue Title

Fixes: #31 

-   [Y] Has the code been formatted?
-   [Y] Have dependencies/packages been added?

### Changes
Enabled CORS in the backend router to allow for API calls made by the frontend from http://localhost:3000. 
Fixed the linting manually, the checkers now pass.
### Testing
-   [Y] Does the code have sufficient automated test coverage? If not, include a reasoning and steps required to manually QA
-   I checked by making multiple requests from the frontend and they all came back with response status 200.

### CODE REVIEW

**Review by jcserv — CHANGES_REQUESTED** [review]
Tested it on my end, looks good just need to fix the linting changes!

**Review by jcserv — APPROVED** [review]

**Comment by krishchow** [comment]
Can you squash these commits?

File: README.md [hunk 1]
```diff
@@ -54,9 +54,11 @@ discussion-board follows Gitflow
 1. `git checkout -b DEV-000` (where 000 corresponds to the issue number)
 2. Write your code
 3. Open a PR from that branch to develop (in the main repo)
-5. Fill out the pull request template accordingly
-6. To be approved, code must have adequate test coverage + formatted properly
+4. Fill out the pull request template accordingly
+5. To be approved, code must have adequate test coverage + formatted properly
+
 - Check frontend formatting with `npm run lint` and `make prettier`
```
Comment (krishchow) [inline README.md:60]: Should this be 6?

File: backend/README.md [hunk 2]
```diff
@@ -82,3 +82,8 @@ specificUserRouter := router.Group("/user/:uid"){
     // and is used for archiving a course in the db, we require the cid for looking up the course in the db, and we can just update the isarc value to be true
 
 }
+
+### CORS
+
+Currently we have given access to http://localhost:3000 as an origin source for making calls to the backend. Wehn pushing the app
```
Comment (krishchow) [inline backend/README.md:88]:
```suggestion
Currently we have given access to http://localhost:3000 as an origin source for making calls to the backend. When pushing the app
```

Comment (krishchow) [reply backend/README.md:88]: I think we can also add https://dialog-gamma.vercel.app/, right @jcserv?

File: backend/httpd/main.go [hunk 3]
```diff
@@ -1,14 +1,33 @@
...
 	r := gin.Default()
+
+	// This allows for front end calls made from the origin source
+	// http://localhost:3000
+	// This will need to be changed/updated to either be "*", ie
+	// allow all origins (which could be bad for security), or
+	// the url for where ever we host our front end.
+	// All RESTful api methods are allowed
+	r.Use(cors.New(cors.Config{
+		AllowOrigins:     []string{"http://localhost:3000"},
```
Comment (krishchow) [inline backend/httpd/main.go:18]:
```suggestion
		AllowOrigins:     []string{"http://localhost:3000", "https://dialog-gamma.vercel.app/"},
```

**Pull request closed without merging**